### PR TITLE
refactor: cleanups from writing docstring

### PR DESCRIPTION
**Pull request by ss2165**
go commit by commit
mostly things like making methods/classes internal

### CODE REVIEW

File: hugr-py/src/hugr/val.py [hunk 1]
```diff
@@ -50,13 +50,16 @@ def bool_value(b: bool) -> Sum:
...
-    def type_(self) -> tys.Tuple:
-        return tys.Tuple(*(v.type_() for v in self.vals))
+    def __init__(self, *vals: Value):
+        val_list = list(vals)
+        super().__init__(
+            tag=0, typ=tys.Tuple(*(v.type_() for v in val_list)), vals=val_list
+        )
 
-    def to_serial(self) -> sops.TupleValue:
+    def to_serial(self) -> sops.TupleValue:  # type: ignore[override]
```
Comment (aborgna-q) [inline hugr-py/src/hugr/val.py:62]: Any idea why the ignore is needed?
It seems straightforward code

**Review by aborgna-q — APPROVED** [review]
Nice cleanup 👍

File: hugr-py/src/hugr/val.py (hunk 1 above)
Comment (ss2165) [reply hugr-py/src/hugr/val.py:62]: `sops.TupleValue` isn't an instance of `sops.SumValue` so mypy doesn't like the override of `Sum.to_serial`

Comment (ss2165) [reply hugr-py/src/hugr/val.py:62]: will add comment

**Pull request merged**